backend.py

```diff
@@ -104,17 +120,43 @@ class DepthDataListener(roypy.IDepthDataListener):
     def __init__(self, queue):
         super().__init__()
         self.queue = queue
+        self._ir_ref = None
+
+        self._data_depth = None
+        self._data_ir = None  # type: roypycy.PyIrImage
+
+    def _check_frame(self):
+        if self._data_depth is None or self._data_ir is None:
```

Comment:
This does not guarantee that we match IR and depth frames correctly. Assuming that both are assigned matching timestamps if they were captured at the same time, we need to additionally check for the equality of their timestamps before putting them into the queue.

Reply:
I will add this as an issue as a future todo.